### PR TITLE
[TT-10633] Add GraphQLIntrospectionConfig struct to ApiDefinition CRD

### PR DESCRIPTION
**Pull request by buraksekili**
This PR reverts changes of graphql introspection PR #669.
## Related Issue

https://tyktech.atlassian.net/browse/TT-10633

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](https://github.com/TykTechnologies/tyk-operator/blob/master/CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] If you've changed API models, please update CRDs.
  - [ ] `make manifests`
  - [ ] `make helm`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: api/model/api.go [hunk 1]
```diff
@@ -1163,6 +1163,13 @@ type GraphQLConfig struct {
 
 	// Supergraph holds the configuration for a GraphQL federation supergraph.
 	Supergraph GraphQLSupergraphConfig `json:"supergraph,omitempty"`
+
+	// Introspection holds the configuration for GraphQL Introspection
+	Introspection GraphQLIntrospectionConfig `json:"introspection,omitempty"`
+}
+
+type GraphQLIntrospectionConfig struct {
+	Disabled bool `json:"disabled,omitempty"`
```
Comment (komalsukhani) [inline api/model/api.go:1172]: Can we set default value to true?

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: api/model/api.go (hunk 1 above)
Comment (buraksekili) [reply api/model/api.go:1172]: @komalsukhani  on second thought, I propose setting this value to false by default, as it aligns with users' expectations to see the result of GraphQL introspection by default. For those who prefer an alternative to disable introspection, a configuration option is available on Tyk side to disable it. Opting for a default of true might potentially lead to confusion. What are your thoughts on this approach?

Comment (komalsukhani) [reply api/model/api.go:1172]: Won't it cause problem as API Definition will not match with CRD?

Comment (buraksekili) [reply api/model/api.go:1172]: I do not think that as APIDefinition on Tyk is only compared with the hash of ApiDefinition on Tyk that is obtained after creating CRD.

Comment (komalsukhani) [reply api/model/api.go:1172]: okay. We can confirm that during testing.

**Review by komalsukhani — APPROVED** [review]

**Pull request merged**